py/server/deephaven/experimental/s3.py

```diff
@@ -23,6 +23,91 @@
     included in the package. This is an opt-out functionality included by default. If not included, importing this
     module will fail to find the java types.
 """
+
+class Credentials(JObjectWrapper):
+    """
+    Credentials object for authenticating with S3 server.
+    """
+    j_object_type = _JCredentials
+
+    def __init__(self, _j_object: jpy.JType):
+        """
+        Initializes the credentials object.
+
+        Args:
+            _j_object (Credentials): the Java credentials object.
+        """
+        self._j_object = _j_object
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self._j_object
+
+    @classmethod
+    def resolving(cls) -> 'Credentials':
+        """
+        Default credentials provider used by Deephaven which resolves credentials in the following order:
+        1. If a profile name, config file path, or credentials file path is provided, use ProfileCredentialsProvider.
+        Ref: https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/auth/credentials/ProfileCredentialsProvider.html
+
+        2. If not, check all places mentioned in DefaultCredentialsProvider and fall back to AnonymousCredentialsProvider.
```

Comment:
This is referencing a bunch of Java stuff.  Not good python documentation.  We need to spell out what happens.